app/src/main/java/com/example/androidtrainingproject/ui/home/HomeViewModel.kt

```diff
@@ -51,11 +57,52 @@ class HomeViewModel @Inject constructor(
         val newFilteredProducts = allProducts.filter { product: ProductResponse ->
             product.title.contains(newValue, ignoreCase = true)
         }
-        filteredProducts = newFilteredProducts
+        filteredProducts.clear()
+        filteredProducts.addAll(newFilteredProducts)
     }
 
     fun clearSearchText() {
         _searchText.value = ""
-        filteredProducts = allProducts
+        filteredProducts.clear()
+        filteredProducts.addAll(allProducts)
+    }
+
+    fun applyFilters(
+        checkedCategories: MutableList<String>,
```

Comment:
All of those parameters belong to the view model anyway so they shouldn't be passed to the function.